### ISSUE TITLE
p: UInt = -1;

### ISSUE THREAD

**Issue by nanjizal** [HaxeFoundation/haxe#7822]
given ts have just posted a Positive typing.
https://twitter.com/GiulioCanti/status/1098173442019463168
really Haxe should error this, I understand it may not be able to pick up all UInt failures - but this one I would have expected?
http://try-haxe.mrcdk.com/#4A3D0

**Comment by nanjizal** [comment]
I can look into the abstract used but I presume we need this not to impact runtime.

**Comment by Simn** [comment]
This has defined semantics and shouldn't be an error.

And wtf is a brand combinator...

**Comment by Simn** [comment]
Actually it doesn't really have defined semantics because UInt has no defined size.

**Comment by nanjizal** [comment]
UInt trace does not work on haxe 3.4.3 ( or number that the link links to ) but that is not really a relevant but just interesting historical detail.

I understand that improving UInt is not simple without overhead. It's not a feature that bothers me, on as3 I found Vector of UInt carried overhead that I could not justify.  I rarely consider using UInt, it's more common to create abstracts for limited number sets like Radians that loop back so ideally improvements would take into account number/set theory rather than provide superficial security.

So will close - I only opened in case it was simple to improve from the compiler - I am sure there are abstract number implementations for UInt that can provided for expected behaviour, but at cost. Thanks for thinking about it.
